nexus/src/app/vpc_router.rs

```diff
@@ -162,24 +138,50 @@ impl super::Nexus {
 
     // Routes
 
-    #[allow(clippy::too_many_arguments)]
+    pub fn vpc_router_route_lookup<'a>(
+        &'a self,
+        opctx: &'a OpContext,
+        route_selector: &'a params::RouteSelector,
+    ) -> LookupResult<lookup::RouterRoute<'a>> {
+        match route_selector {
+            params::RouteSelector {
+                route: NameOrId::Id(id),
+                router_selector: None,
+            } => {
+                let route = LookupPath::new(opctx, &self.db_datastore)
+                    .router_route_id(*id);
+                Ok(route)
+            }
+            params::RouteSelector {
+                route: NameOrId::Name(name),
+                router_selector: Some(selector),
+            } => {
+                let route = self
+                    .vpc_router_lookup(opctx, selector)?
+                    .router_route_name(Name::ref_cast(name));
+                Ok(route)
+            }
+            params::RouteSelector {
+                route: NameOrId::Id(_),
+                router_selector: Some(_),
+            } => Err(Error::invalid_request(
+                "when providing subnet as an ID, vpc should not be specified",
```

Comment:
This looks like a copy-paste from the updates to the VPC Subnet?

Reply:
Yep! good catch.

Reply:
FYI, looks like this one didn't get fixed up before the merge. Probably want to fix it up later.

Reply:
Yeah, I'll make a docs sweep.